services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -36,7 +36,26 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     }
   }
   override def getFiles(project: Project, path: Path)(implicit ec: ExecutionContext): AsyncResult[List[String]] = ???
-  override def getProjectVersions(project: Project)(implicit ec: ExecutionContext): AsyncResult[Project] = ???
+
+  override def getProjectVersions(project: Project)(implicit ec: ExecutionContext): AsyncResult[Seq[Version]] =
+    if (project.repository == null)
+      Future.failed(VersioningException("There is no repository in this project"))
```

Comment:
As we discussed, you should not check it here. If you do have a `Project`, you may assume that all of his fields are filled correctly. Otherwise, it must be fixed by whoever created an invalid object.

Reply:
Fixed. Added creation of project on adding to DB.